contracts/contracts/coordination/ManagedAllowList.sol

```diff
@@ -2,149 +2,119 @@
+        initializeCohortAdminRole(ritualId);
+        grantRole(authAdminRole(ritualId), account);
     }
 
     /**
-     * @notice Sets the administrator caps for a ritual
-     * @dev Only active rituals can set administrator caps
+     * @notice Checks if the sender is an Authorization Admin of the ritual
      * @param ritualId The ID of the ritual
-     * @param addresses The addresses of the administrators
-     * @param value The cap value
      */
-    function setAdministratorCaps(
-        uint32 ritualId,
-        address[] calldata addresses,
-        uint256 value
-    ) internal {
-        require(
-            coordinator.isRitualActive(ritualId),
-            "Only active rituals can set administrator caps"
-        );
-        for (uint256 i = 0; i < addresses.length; i++) {
-            allowance[LookupKey.lookupKey(ritualId, addresses[i])] = value;
-            emit AdministratorCapSet(ritualId, addresses[i], value);
-        }
-        authActions[ritualId] += addresses.length;
+    modifier canSetAuthorizations(uint32 ritualId) virtual override {
+        require(hasRole(authAdminRole(ritualId), msg.sender), "Only auth admin is permitted");
+        _;
     }
 
     /**
-     * @notice Adds administrators for a ritual
+     * @notice Authorizes a list of addresses for a ritual
      * @param ritualId The ID of the ritual
-     * @param addresses The addresses of the administrators
-     * @param cap The cap value
+     * @param addresses The addresses to be authorized
      */
-    function addAdministrators(
+    function authorize(
         uint32 ritualId,
-        address[] calldata addresses,
-        uint256 cap
-    ) external onlyCohortAuthority(ritualId) {
-        setAdministratorCaps(ritualId, addresses, cap);
+        address[] calldata addresses
+    ) external override canSetAuthorizations(ritualId) {
+        setAuthorizations(ritualId, addresses, true);
+        for (uint256 i = 0; i < addresses.length; i++) {
+            bytes32 lookupKey = LookupKey.lookupKey(ritualId, addresses[i]);
+            authAdmins[msg.sender][lookupKey] = true;
+        }
     }
 
     /**
-     * @notice Removes administrators for a ritual
+     * @notice Deauthorizes a list of addresses for a ritual
      * @param ritualId The ID of the ritual
-     * @param addresses The addresses of the administrators
+     * @param addresses The addresses to be deauthorized
      */
-    function removeAdministrators(
+    function deauthorize(
         uint32 ritualId,
         address[] calldata addresses
-    ) external onlyCohortAuthority(ritualId) {
-        setAdministratorCaps(ritualId, addresses, 0);
+    ) external override canSetAuthorizations(ritualId) {
+        for (uint256 i = 0; i < addresses.length; i++) {
+            bytes32 lookupKey = LookupKey.lookupKey(ritualId, addresses[i]);
+            require(
+                authAdmins[msg.sender][lookupKey],
+                "Encryptor must be authorized by the sender first"
```

Comment:
```suggestion
                "Encryptor has not been previously authorized by the sender"
```

WDYT? IMHO the original message seems to express that there previous step to do before deauthorize, instead of expressing the reason of why the deauthorization can not be done.

Reply:
a bit long but everything longer than 32 bytes are not good anyway so I'm ok with suggestion